lib/dotnot/dotnot_test.go

```diff
@@ -0,0 +1,78 @@
+package dotnot
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+var testsuite = []struct {
+	name string
+	from string
+	to   string
+}{
+	{
+		name: "Do not change single root key",
+		from: `{"root":"key"}`,
+		to:   `{"root":"key"}`,
+	},
+	{
+		name: "Do not change multiple root keys",
+		from: `{"another":"key","root":"key"}`,
+		to:   `{"another":"key","root":"key"}`,
+	},
+	{
+		name: "Dotnotate a single level",
+		from: `{"nesting":{"one":true},"root":"key"}`,
+		to:   `{"nesting.one":true,"root":"key"}`,
+	},
+	{
+		name: "Dotnotate a bunch of levels",
+		from: `{"nest":{"two":{"three":{"four":"layers_deep"}},"two_half":{"and_siblings_too":true}},"root":"key"}`,
+		to:   `{"nest.two.three.four":"layers_deep","nest.two_half.and_siblings_too":true,"root":"key"}`,
+	},
+	{
+		name: "Dotnotate arrays safely",
+		from: `{"nest":{"two":["arrays","must","make","it","through"],"two_half":{"and_siblings_too":true}},"root":"key"}`,
+		to:   `{"nest.two":["arrays","must","make","it","through"],"nest.two_half.and_siblings_too":true,"root":"key"}`,
```

Comment:
What will it do with arrays containing objects?

Reply:
You're testing my skills right? Don't think I won't rise to the challenge

Reply:
https://github.com/cuvva/cuvva-public-go/pull/18/commits/109363f752f0b77b317527338e50ad87ae8b3901

Reply:
oof, very nice 👌